### PR TITLE
Use fsPath instead of path for config file

### LINKED ISSUE #57: Extension configured with malformed path when discovering flow.json in workspace
#### VSCode
Version: 1.58.2 (system setup)
Commit: c3f126316369cd610563c75b1b1725e0679adfb3
Date: 2021-07-14T22:10:15.214Z
Electron: 12.0.13
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19043

#### Flow
flow-cli:
Version: v0.26.0
Commit: 87011245afd3fb98ebab2533cc05e1d5a56410e5

vscode-cadence: 
Version: 0.6.3

### Problem
When opening a cadence file (triggering the language server to start), an error occurs stating the filename, directory name, or volume label syntax is incorrect for the file flow.json.

cls-bad-path

When starting the flow emulator using the status bar or command palette the statement passed to the terminal includes the same malformed path:

`flow project start-emulator --config-path=/c:/Source/.../kitty-items/flow.json --verbose`

In this example, the kitty-items repo is opened as a folder/workspace in vscode.

### Possible solution

Changing the following line to use the `fsPath` attribute (rather than the `path` attribute) on the `Uri` object, seems to have corrected this issue in my test environment.

https://github.com/onflow/vscode-cadence/blob/4c3f06fedc649da4a65a519c1fd38c8fe226aa8c/src/config.ts#L42

### Context

Currently working on setting up flow developer workstation and reviewing example code from various repos.

### PR DESCRIPTION
**Pull request by eburnette**
The problem was that a config file path like
'/c:/something' was being used instead of the
correct path 'c:\something\'.
This is the solution suggested by @fluxium.

Unfortunately a number of other changes were
necessary to make the extension build again
with the latest packages on Windows. Then there
were a number of new lint messages checked, and
new messages because of a newer typescript.
I think I got them all fixed except the cypress test
is failing due to some missing monaco element.
This is likely caused by a dom change in a new version
that needs to be accomodated in the test case.

Closes #57 

## Description

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by eburnette** [comment]
Sorry I forgot to mention:
Fixes https://github.com/onflow/vscode-cadence/issues/57

File: README.md [hunk 1]
```diff
@@ -1,9 +1,5 @@
-<br />
+<h1>Cadence Visual Studio Code Extension</h1>
 <p align="center">
-  <a href="https://docs.onflow.org/vscode-extension/">
-    <img src="./vscode-banner.svg" alt="Logo" width="600" height="auto">
-  </a>
-
```
Comment (sideninja) [inline README.md:6]: Why was this removed?

File: package.json [hunk 2]
```diff
@@ -3,29 +3,28 @@
...
-    "vscode:prepublish": "npm run -S esbuild-base -- --minify",
-    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/main.js --external:vscode --format=cjs --platform=node",
-    "esbuild": "npm run -S esbuild-base -- --sourcemap",
-    "esbuild-watch": "npm run -S esbuild-base -- --sourcemap --watch",
+    "build": "tsc -p ./",
+    "watch": "tsc -watch -p ./",
     "unit-test": "ts-mocha **/*.unit.test.ts",
-    "integration-test": "npm run esbuild && tsc -p ./ && node ./out/test/run-test.js",
-    "package": "./node_modules/.bin/vsce package",
-    "package-test": "./node_modules/.bin/vsce package --out ./src/test/fixtures/workspace/cadence.vsix",
```
Comment (sideninja) [inline package.json:19]: should `npx` be used?

File: package.json [hunk 3]
```diff
@@ -3,29 +3,28 @@
...
   "description": "This extension integrates Cadence, the resource-oriented smart contract programming language of Flow, into Visual Studio Code.",
-  "version": "0.6.3",
+  "version": "0.6.4",
   "repository": {
     "type": "git",
     "url": "https://github.com/onflow/vscode-cadence.git"
   },
   "scripts": {
-    "vscode:prepublish": "npm run -S esbuild-base -- --minify",
-    "esbuild-base": "esbuild ./src/extension.ts --bundle --outfile=out/main.js --external:vscode --format=cjs --platform=node",
```
Comment (sideninja) [inline package.json:13]: wouldn't pre-publish be needed anymore when doing a release? also esbuild is not needed?

File: README.md (hunk 1 above)
Comment (eburnette) [reply README.md:6]: SVG's are no longer allowed in the readme. Apparently, it's a security hole.

File: package.json (hunk 2 above)
Comment (eburnette) [reply package.json:19]: It could be... I didn't want to change this more than it already was.

File: package.json (hunk 3 above)
Comment (eburnette) [reply package.json:13]: It looks like the author did all this to make sure extension.ts was built into main.js. It was just easier to specify extension.js as the entry point. Now it's more in line with what the recommended "hello, world" extension uses. I ran the instructions in https://code.visualstudio.com/api/get-started/your-first-extension and followed their advice when possible.

File: README.md (hunk 1 above)
Comment (sideninja) [reply README.md:6]: would another image format work? I would like to keep the banner image since all the tools have one.

File: package.json (hunk 2 above)
Comment (sideninja) [reply package.json:19]: But will this work without having the .bin folder set in PATH?

File: package.json (hunk 3 above)
Comment (sideninja) [reply package.json:13]: That's great that you follow the recommended patterns. I'm just curious if the minification will still happen since I think esbuild was used to minify the extension size.

File: README.md (hunk 1 above)
Comment (eburnette) [reply README.md:6]: Yes, if you want to give me a .png I could use that.

File: package.json (hunk 2 above)
Comment (eburnette) [reply package.json:19]: vsce and tsc will have to be on the path, and it will work for both Windows and Linux. The current syntax only works on Linux. On Windows it gives an error about an invalid path.

    > cadence@0.6.4 package
    > ./node_modules/.bin/vsce package
    
    '.' is not recognized as an internal or external command,
    operable program or batch file.

File: package.json (hunk 3 above)
Comment (eburnette) [reply package.json:13]: Good point, I'll go back to using esbuild to get the minification.

File: README.md (hunk 1 above)
Comment (eburnette) [reply README.md:6]: If you look at https://github.com/onflow/vscode-cadence in github dark mode, the text is hard to see. Also I looked at several other extensions and none of the ones I randomly picked use an image, they just use a header text. So I'd recommend just sticking with text.

Comment (sideninja) [reply README.md:6]: We can deal with this as finish touches, don't worry about it, and you have good point about the dark mode.

**Comment by eburnette** [comment]
How do I fix this synk error:

    `snyk` requires an authenticated account. Please run `snyk auth` and try again.

**Comment by sideninja** [comment]
> How do I fix this synk error:
> 
> ```
> `snyk` requires an authenticated account. Please run `snyk auth` and try again.
> ```
> 
> Upon further investigation, it looks like "snyk test" requires an auth token, but "snyk protect" does not. The auth token requirement was added at some point, and the package.json was using an old version. When I updated it, I ran into this problem. I will revert to the old version of snyk, but really, the best solution is to provide the auth token.

Yes, that can be resolved, but maybe it's better to revert now and not be blocked by it and resolve that as part of later PR

**Comment by eburnette** [comment]
> Yes, that can be resolved, but maybe it's better to revert now and not be blocked by it and resolve that as part of later PR

It's done, how do I make it re-run the CI action?

**Comment by sideninja** [comment]
> > Yes, that can be resolved, but maybe it's better to revert now and not be blocked by it and resolve that as part of later PR
> 
> It's done, how do I make it re-run the CI action?

The CI actions won't be run anymore since I've changed the base branch to be a non-master branch. You can run tests, linting and synk check locally. CI will be run after this is merged into base (`submissions/issue-3/koala`) branch.

**Comment by eburnette** [comment]
I ran tests, lint, and sync check locally. Do you need me to do anything else?

**Comment by sideninja** [comment]
> I ran tests, lint, and sync check locally. Do you need me to do anything else?

No, no feel free to merge this, and for future reference I think it will be easier that you go and merge this PRs yourself (be sure to target same base branch as this PR) and then only open a PR after the milestone is complete, so we don't block each other.

**Comment by eburnette** [comment]
I don't have permission to merge pull requests here.

**Review by 10thfloor — COMMENTED** [review]
lgtm

**Comment by eburnette** [comment]
Ok I must be going about this wrong. My intent was to submit a pull request to get a change into the master branch so that people can develop the extension on windows.

**Comment by sideninja** [comment]
Are you trying to make this part of the FLIP fest? In case you are, this should stay out of master until the work is complete, because else anyone else working on this will conflict with your work. The idea is that after all milestones are completed this potentially is merged into master.

**Comment by eburnette** [comment]
No this is not a flip-fest problem, I just needed it to be done in order to do one of the flip-fest problems. Thanks.

**Comment by figs999** [comment]
The bug this fixes also blocks me from working with VSCode on windows. Merging this and getting a new build of the VSCode plugin would be super helpful.

**Comment by sideninja** [comment]
@eburnette Ok, I see, I will test the changes and merge them as soon as possible. You can continue your work by basing the branch on top of this one.

File: package.json (hunk 2 above)
Comment (dazhilang018900) [reply package.json:19]: you can run this cmd '.\node_modules\.bin\vsce package ' in your workdir
